### PR TITLE
fix(eslint): fix eslint-plugin build, test and lint

### PR DESCRIPTION
**Pull request by matthieu-crouzet**

### CODE REVIEW

File: packages/@o3r/eslint-plugin/src/rules/json/json-dependency-versions-harmonize/json-dependency-versions-harmonize.spec.ts [hunk 1]
```diff
@@ -1,25 +1,18 @@
...
   languageOptions: {
-    parser: jsonParser,
-    parserOptions: {
-      ecmaVersion: 2018,
-      sourceType: 'module'
-    }
+    parser: jsonParser
   }
-});
+} as any);
```
Comment (matthieu-crouzet) [inline packages/@o3r/eslint-plugin/src/rules/json/json-dependency-versions-harmonize/json-dependency-versions-harmonize.spec.ts:15]: I don't understand why the any is needed here... but without it's failing because it says that `languageOptions` does not exist on `FlatConfig.Config` (which is false, it does)

File: packages/@o3r/eslint-plugin/src/rules/json/json-dependency-versions-harmonize/json-dependency-versions-harmonize.spec.ts [hunk 2]
```diff
@@ -113,23 +95,41 @@ ruleTester.run('json-dependency-versions-harmonize', jsonDependencyVersionsHarmo
...
+          ]
         }
       ]
     },
     {
       filename: packageToLint,
       output: JSON.stringify({ peerDependencies: { myOtherDep: '~2.0.0' } }),
       code: JSON.stringify({ peerDependencies: { myOtherDep: '^2.0.0' } }),
-      options: [{ alignPeerDependencies: true }],
+      options: [{ alignPeerDependencies: true } as VersionsHarmonizeOptions],
```
Comment (matthieu-crouzet) [inline packages/@o3r/eslint-plugin/src/rules/json/json-dependency-versions-harmonize/json-dependency-versions-harmonize.spec.ts:115]: Needed else the following tests expect to have the same options type, I don't understand why because typings seems okay...

File: packages/@o3r/eslint-plugin/src/rules/yaml/yarnrc-package-extensions-harmonize/yarnrc-package-extensions-harmonize.spec.ts [hunk 3]
```diff
@@ -1,24 +1,19 @@
...
-
 const ruleTester = new RuleTester({
   languageOptions: {
     parser: yamlParser,
     parserOptions: {
       defaultYAMLVersion: '1.2'
     }
   }
-});
+} as any);
```
Comment (matthieu-crouzet) [inline packages/@o3r/eslint-plugin/src/rules/yaml/yarnrc-package-extensions-harmonize/yarnrc-package-extensions-harmonize.spec.ts:16]: I don't understand why the any is needed here... but without it's failing because it says that `languageOptions` does not exist on `FlatConfig.Config` (which is false, it does)

File: .yarnrc.yml [hunk 4]
```diff
@@ -18,6 +18,15 @@ packageExtensions:
...
       "@swc/types": "*"
+  "@typescript-eslint/rule-tester@*":
+    dependencies:
+      "@typescript-eslint/parser": "~8.12.2"
+  "angular-eslint@*":
+    dependencies:
+      "@typescript-eslint/utils": ~8.12.2
+  "@angular-eslint/eslint-plugin-template@*":
+    dependencies:
+      "@typescript-eslint/types": "^8.0.0"
```
Comment (matthieu-crouzet) [inline .yarnrc.yml:29]: This dep is declared as peerDep but missing in `angular-eslint`
I tried the following code but it still complains, that's the only way I found to fix it
```yml
"angular-eslint@*":
    dependencies:
      "@typescript-eslint/utils": ~8.12.2      
      "@typescript-eslint/types": "^8.0.0"

File: packages/@o3r/eslint-plugin/src/rules/template/template-async-number-limitation/template-async-number-limitation.spec.ts [hunk 5]
```diff
@@ -24,9 +26,9 @@ ruleTester.run(name, templateAsyncNumberLimitation as any /* workaround for 5.9.
     convertAnnotatedSourceToFailureCase({
       description: 'should fail when there are more than 2 async pipes on a single HTML element',
       annotatedSource: `
-        <div ngIf="obs$ | async | async | async"></div>
-        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-      `,
+<div ngIf="obs$ | async | async | async"></div>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+      `.trim(),
```
Comment (matthieu-crouzet) [inline packages/@o3r/eslint-plugin/src/rules/template/template-async-number-limitation/template-async-number-limitation.spec.ts:31]: added `.trim()` in several places to avoid issue due to indentation.

File: apps/chrome-devtools/package.json [hunk 6]
```diff
@@ -50,10 +46,8 @@
     "@types/chrome": "^0.0.271",
     "@types/jest": "~29.5.2",
     "@types/tinycolor2": "^1.4.6",
-    "@typescript-eslint/eslint-plugin": "^8.0.0",
-    "@typescript-eslint/parser": "^8.0.0",
-    "@typescript-eslint/types": "^8.0.0",
-    "@typescript-eslint/utils": "^8.0.0",
+    "@typescript-eslint/parser": "~8.12.2",
```
Comment (matthieu-crouzet) [inline apps/chrome-devtools/package.json:49]: Already present in `typescript-eslint` but some ESLint plugin didn't migrate yet to this package

File: package.json [hunk 7]
```diff
@@ -26,7 +26,7 @@
     "test:affected": "yarn nx affected --target=test --parallel $(yarn get:cpus-number) --cacheDirectory=$(yarn get:current-dir)/.cache/jest",
     "test-e2e": "yarn nx run-many --target=test-e2e --parallel $(yarn get:cpus-number)",
     "test-int": "yarn nx run-many --target=test-int --parallel $(yarn get:cpus-number)",
-    "postinstall": "husky && yarn build:lint && yarn harmonize:version && yarn update-yarn-sdks",
+    "postinstall": "husky && yarn build:lint && yarn update-yarn-sdks",
```
Comment (fpaul-1A) [inline package.json:29]: we want to remove this?

Comment (matthieu-crouzet) [reply package.json:29]: oops :D

**Review by cpaulve-1A — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

File: packages/@o3r/eslint-plugin/tsconfig.spec.json [hunk 8]
```diff
@@ -1,5 +1,9 @@
 {
   "extends": "../../../tsconfig.jest",
+  "compilerOptions": {
+    "moduleResolution": "NodeNext",
```
Comment (kpanot) [inline packages/@o3r/eslint-plugin/tsconfig.spec.json:4]: Why did you need it, it is generally not compatible with TS reviver (as it inforce to add the `.js` in the imports in the code :S

File: packages/@o3r/eslint-plugin/tsconfig.spec.json [hunk 9]
```diff
@@ -1,5 +1,9 @@
...
+    "module": "NodeNext"
```
Comment (kpanot) [inline packages/@o3r/eslint-plugin/tsconfig.spec.json:5]: Why not keeping the default from `base`
